lib/langchain/assistant/messages/mistral_ai_message.rb

```diff
@@ -46,29 +46,10 @@ def llm?
         # @return [Hash] The message as an MistralAI API-compatible hash
         def to_hash
```

Comment:
What're your thoughts on doing something like this:
```ruby
def to_hash
  if assistant?
    to_assistant_hash
  elsif tool?
    to_tool_hash
  ...
end

def to_assistant_hash
  h = {
    role: role,
    content: to_content
  }
  h[:tool_calls] = tool_calls if tool_calls.any?
end
```

Reply:
It's pretty similar to what you've got just slightly different stylistically.

Reply:
Hmm, I am very on the fence about it as one hash or hash by role. 

I think declaring the separation by role makes sense since it feels like Mistral has declared separate API Contracts by role (as we see with `content`) 😄 

I updated it to do it by role and refactored the content hash so it was just building the array value.